### DISCUSSION TITLE
A Question About Weight Format

### DISCUSSION THREAD

**Discussion by zyt1024** [fastmachinelearning/hls4ml#871] [General]
Hello! In the implementation of `pointwise_conv_2d_cl `in `hls4ml,` for` a 1x1 convolutional kernel`, is the weight format` (Cout, Cin)` or`( Cin, Cout) `in storage?     I think is (Cout,Cin), is right?

**Comment by vloncar** [comment]
Depends on the `Strategy`, in case of `latency` strategy for 1x1 kernel it is `(Cin, Cout)` but for resource it is `(Cout, Cin)`. More generally, for a `kernel_height x kernel_width` Conv2D the layout of weights is `(kernel_height, kernel_width, Cin, Cout)` for `latency` strategy, and `(Cout, kernel_height, kernel_width, Cin)` for `resource` strategy.

**Reply by zyt1024** [reply]
ok,Thanks,I understand.
